modules/extensions/calico.tf

```diff
@@ -16,16 +16,43 @@ resource "null_resource" "install_calico" {
 
   depends_on = [null_resource.install_k8stools_on_operator, null_resource.write_kubeconfig_on_operator]
 
+  provisioner "remote-exec" {
+    inline = [
+      "mkdir -p ${var.calico_staging_dir}"
+    ]
+  }
+
+  provisioner "file" {
+    source      = "${path.module}/scripts/yaml"
+    destination = "${var.calico_staging_dir}/"
+  }
+
   provisioner "file" {
-    content     = local.install_calico_template
-    destination = "/home/opc/install_calico.sh"
+    source      = "${path.module}/scripts/split_yaml.awk"
+    destination = "${var.calico_staging_dir}/split_yaml.awk"
+  }
+
+  provisioner "file" {
+    content     = local.calico_env_template
+    destination = "${var.calico_staging_dir}/calico_env.sh"
+  }
+
+  provisioner "file" {
+    source      = "${path.module}/scripts/calico_install.sh"
+    destination = "${var.calico_staging_dir}/calico_install.sh"
   }
 
   provisioner "remote-exec" {
     inline = [
-      "if [ -f \"$HOME/install_calico.sh\" ]; then bash \"$HOME/install_calico.sh\"; rm -f \"$HOME/install_calico.sh\";fi",
+      "chmod +x ${var.calico_staging_dir}/calico_install.sh",
+      "${var.calico_staging_dir}/calico_install.sh",
+      "rm -r ${var.calico_staging_dir}"
```

Comment:
```suggestion
      bash "${var.calico_staging_dir}/calico_install.sh && rm -r ${var.calico_staging_dir}"
```

Reply:
This avoids making the script executable and will only run the `rm` if the script exits cleanly.